### PR TITLE
Dev kubra

### PR DESCRIPTION
**Pull request by NurKubra**
user-entity and user-role created, seuccesMessages class created

### CODE REVIEW

File: src/main/java/com/rentacar/entity/abstracts/User.java [hunk 1]
```diff
@@ -1,21 +1,55 @@
...
+
+    @Column(length = 120, nullable = false)
+    private String password;
+
+    @Column(length = 50, nullable = false)
+    private String birthplace;
+
+    @Column(length = 50, nullable = false)
+    private String birthday;
+
```
Comment (eneskeskin4934) [inline src/main/java/com/rentacar/entity/abstracts/User.java:46]: @JsonIgnore
@ManyToMany
@JoinTable(name = "user_roles_table",
            joinColumns = @JoinColumn(name = "user_id"), //icinde bulundugumuz class
            inverseJoinColumns = @JoinColumn(name = "userRole_id")). //Karsi class

File: src/main/java/com/rentacar/entity/concretes/user/UserRole.java [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+public class UserRole {   //enumlarla aradaki ilskiyi saglayacak class
+
+    @Id
+    @GeneratedValue(strategy = GenerationType.IDENTITY)
+    private Long id;
+
+    @Enumerated(EnumType.STRING)
+    @Column(length = 20)
+    private RoleType roleType;
+
```
Comment (eneskeskin4934) [inline src/main/java/com/rentacar/entity/concretes/user/UserRole.java:29]: @ManyToMany(mappedBy = "UserRoles", fetch = FetchType.LAZY)
private List<User> userList;

File: src/main/java/com/rentacar/entity/abstracts/User.java (hunk 1 above)
Comment (eneskeskin4934) [reply src/main/java/com/rentacar/entity/abstracts/User.java:46]: Bildschirmfoto 2023-10-12 um 13 25 52
Bildschirmfoto 2023-10-12 um 13 30 33

Comment (eneskeskin4934) [reply src/main/java/com/rentacar/entity/abstracts/User.java:46]: Projede gecen manyToMany iliskilerinin ekran goruntusunu paylastim

File: src/main/java/com/rentacar/entity/abstracts/User.java [hunk 3]
```diff
@@ -1,21 +1,55 @@
...
 @AllArgsConstructor
 @NoArgsConstructor
 @SuperBuilder
 public class User {
+    @Id
+    @GeneratedValue(strategy = GenerationType.IDENTITY)
+    private Long id;
 
+    @NotNull(message = "Username mustn't be empty")
+    @Column(length = 50, nullable = false)
```
Comment (eneskeskin4934) [inline src/main/java/com/rentacar/entity/abstracts/User.java:27]: @Column(length = 50, nullable = false, unique = true)

File: src/main/java/com/rentacar/entity/concretes/user/UserRole.java [hunk 4]
```diff
@@ -0,0 +1,42 @@
...
+import javax.persistence.*;
+import java.util.List;
+
+@Entity
+@Table(name ="roles")
+@Data
+@AllArgsConstructor
+@NoArgsConstructor
+@Builder  //toBuilder olmali mi ?
```
Comment (eneskeskin4934) [inline src/main/java/com/rentacar/entity/concretes/user/UserRole.java:19]: Bence toBuilder'a gerek yok hocam

File: src/main/java/com/rentacar/entity/abstracts/User.java [hunk 5]
```diff
@@ -1,21 +1,55 @@
...
+    private String username;
 
+    @Column(length = 50, nullable = false)
+    private String name;
+
+    @Column(length = 50, nullable = false)
+    private String surname;
+
+    @Column(length = 120, nullable = false)
```
Comment (eneskeskin4934) [inline src/main/java/com/rentacar/entity/abstracts/User.java:37]: @JsonProperty(access = JsonProperty.Access.WRITE_ONLY) //hassas veri oldugu icin okuma islemlerinde kullanilmasin

File: src/main/java/com/rentacar/entity/abstracts/User.java [hunk 6]
```diff
@@ -1,21 +1,55 @@
...
+
+    @Column(length = 50, nullable = false)
+    private String birthday;
+
+    @ManyToMany(mappedBy = "roles", fetch = FetchType.LAZY)
+    @JsonIgnore
+    @JoinTable(name = "user_roles_table",
+            joinColumns = @JoinColumn(name = "userRole_id"),
+            inverseJoinColumns = @JoinColumn(name = "user_id"))
+    private Set<UserRole> roles = new HashSet<>();
```
Comment (eneskeskin4934) [inline src/main/java/com/rentacar/entity/abstracts/User.java:62]: Bunlar da eklenebilir
    @Column(unique = true)
    private String phoneNumber;

    @Enumerated(EnumType.STRING)
    private Gender gender;

**Review by eneskeskin4934 — CHANGES_REQUESTED** [review]
Hocam bir kac yere yorum yazdim

File: src/main/java/com/rentacar/entity/abstracts/User.java [hunk 7]
```diff
@@ -1,21 +1,55 @@
...
 @NoArgsConstructor
 @SuperBuilder
 public class User {
+    @Id
+    @GeneratedValue(strategy = GenerationType.IDENTITY)
+    private Long id;
 
+    @NotNull(message = "Username mustn't be empty")
+    @Column(length = 50, nullable = false)
+    private String username;
```
Comment (eneskeskin4934) [inline src/main/java/com/rentacar/entity/abstracts/User.java:29]: (length = 50, nullable = false, unique = true)

File: src/main/java/com/rentacar/entity/abstracts/User.java (hunk 5 above)
Comment (eneskeskin4934) [inline src/main/java/com/rentacar/entity/abstracts/User.java:37]: @JsonProperty(access = JsonProperty.Access.WRITE_ONLY) //hassas veri oldugu icin okuma islemlerinde kullanilmasin

File: src/main/java/com/rentacar/entity/abstracts/User.java (hunk 3 above)
Comment (eneskeskin4934) [inline src/main/java/com/rentacar/entity/abstracts/User.java:27]: @column(length = 50, nullable = false, unique = true) //unique eklenebilir

File: src/main/java/com/rentacar/entity/concretes/user/UserRole.java (hunk 4 above)
Comment (eneskeskin4934) [inline src/main/java/com/rentacar/entity/concretes/user/UserRole.java:19]: Bence toBuilder'a gerek yok hocam

File: src/main/java/com/rentacar/entity/abstracts/User.java (hunk 6 above)
Comment (NurKubra) [reply src/main/java/com/rentacar/entity/abstracts/User.java:62]: Hocam eklemistim aslinda ama gozukmuyor nedense :/

File: src/main/java/com/rentacar/entity/abstracts/User.java (hunk 1 above)
Comment (NurKubra) [reply src/main/java/com/rentacar/entity/abstracts/User.java:46]: dogru hocam duzeltiyorum hemen :)

**Review by eneskeskin4934 — APPROVED** [review]

**Pull request merged**